ci/smoke-test.sh

```diff
@@ -26,7 +27,28 @@ run:
   args: ["smoke"]
 EOF
 
-output=$(./fly --target ci execute --config ./config.yml)
+output=$(./fly --target ci execute --config ./docker-image-config.yml)
+
+if ! echo "${output}" | grep smoke; then
+  echo "Expected to find 'smoke' in output"
+  exit 1
+fi
+
+# be sure the registry-image type works through the mirror config
+cat > registry-image-config.yml << EOF
+platform: linux
+
+image_resource:
+  type: registry-image
+  source:
+    repository: busybox
+
+run:
+  path: echo
+  args: ["smoke"]
+EOF
+
+output=$(./fly --target ci execute --config ./registry-image-config.yml)
```

Comment:
This block has the smoke test run two tasks, one that uses the docker-image resource type and the other registry-image. This should ensure that the registry mirror settings for both resource types are working correctly.